### PR TITLE
all balance and list of validators added

### PR DESCRIPTION
**Pull request by georgeburduhos**

### CODE REVIEW

File: src/screens/pos-actions/pos-action-delegate/delegate-enter-amount/delegate-enter-amount.tsx [hunk 1]
```diff
@@ -82,15 +82,29 @@ export class DelegateEnterAmountComponent extends React.Component<
...
-    public componentDidMount() {
+    public async componentDidMount() {
         this.props.navigation.setParams({ actionText: this.props.actionText });
+
+        const blockchainInstance = getBlockchain(this.props.blockchain);
+        const tokenConfig = getTokenConfig(this.props.blockchain, this.props.token.symbol);
+        blockchainInstance
+            .getStats(this.props.chainId)
+            .getAvailableBalanceForDelegate(this.props.account)
+            .then(data => {
```
Comment (NovacAndrei) [inline src/screens/pos-actions/pos-action-delegate/delegate-enter-amount/delegate-enter-amount.tsx:97]: You can transform this to:

```
try {
    const data = await blockchainInstance.getStats(this.props.chainId).getAvailableBalanceForDelegate(this.props.account);
    ...
} catch(err) {
   ...
}
```

File: src/core/utils/api-client/validators.ts [hunk 2]
```diff
@@ -47,4 +47,46 @@ export class ValidatorsApiClient {
             SentryCaptureException(new Error(JSON.stringify(err)));
         }
     }
+
+    public async fetchDelegatedValidators(account: IAccountState, chainId: string) {
+        try {
+            const response = await this.apiClient.http.post('/wallet-ui/tokenScreen/accountVotes', {
```
Comment (NovacAndrei) [inline src/core/utils/api-client/validators.ts:53]: Here should be `'/walletUi/tokenScreen/accountVotes'`

File: src/screens/token/components/delegate-token/components/tabs/account-tab/account-tab.tsx [hunk 3]
```diff
@@ -83,6 +86,7 @@ export class AccountTabComponent extends React.Component<
             .catch();
```
Comment (NovacAndrei) [inline src/screens/token/components/delegate-token/components/tabs/account-tab/account-tab.tsx:86]: Maybe send some Sentry errors if needed? In order not to have empty catches
If this was not already handled.

File: src/screens/pos-actions/pos-action-quick-delegate/quick-delegate-enter-amount/quick-delegate-enter-amount.tsx [hunk 4]
```diff
@@ -45,23 +49,49 @@ const mapDispatchToProps = {
...
+            amount: undefined
+        };
     }
 
     public componentDidMount() {
         this.props.navigation.setParams({ actionText: this.props.actionText });
+
+        const blockchainInstance = getBlockchain(this.props.blockchain);
+        const tokenConfig = getTokenConfig(this.props.blockchain, this.props.token.symbol);
+        blockchainInstance
```
Comment (krisboit) [inline src/screens/pos-actions/pos-action-quick-delegate/quick-delegate-enter-amount/quick-delegate-enter-amount.tsx:81]: use async/await

File: src/screens/pos-actions/pos-action-redelegate/redelegate-enter-amount/redelegate-enter-amount.tsx [hunk 5]
```diff
@@ -83,15 +83,29 @@ export class RedelegateEnterAmountComponent extends React.Component<
...
+            amount: undefined
         };
     }
 
     public componentDidMount() {
         this.props.navigation.setParams({ actionText: this.props.actionText });
+
+        const blockchainInstance = getBlockchain(this.props.blockchain);
+        const tokenConfig = getTokenConfig(this.props.blockchain, this.props.token.symbol);
+        blockchainInstance
```
Comment (krisboit) [inline src/screens/pos-actions/pos-action-redelegate/redelegate-enter-amount/redelegate-enter-amount.tsx:95]: use async/await

Comment (georgeburduhos) [reply src/screens/pos-actions/pos-action-redelegate/redelegate-enter-amount/redelegate-enter-amount.tsx:95]: The reason i am not using async/await is that the value that is need to get is not mandatory for the screen to display. The balance value can be shown after the entire screen is rendered.

**Pull request merged**